src/main/kotlin/com/pubnub/api/v2/subscriptions/SubscriptionSet.kt

```diff
@@ -0,0 +1,43 @@
+package com.pubnub.api.v2.subscriptions
+
+import com.pubnub.api.v2.callbacks.EventEmitter
+
+abstract class SubscriptionSet internal constructor() : EventEmitter, SubscribeCapable, AutoCloseable {
```

Comment:
I have following test that doesn't pass. I am not sure if test contains issues or there is a gap somewhere in the code.

```
    @Test
    fun testSubscriptionSet(){
        val success = AtomicBoolean()
        val expectedMessage = randomValue()
        val randomChannelName01 = "myChannel01"
        val randomChannelName02 = "myChannel02"
        val subscriptionSetOf: SubscriptionSet = pubnub.subscriptionSetOf(
            channels = setOf(randomChannelName01, randomChannelName02),
            options = ChannelOptions.receivePresenceEvents()
        )

        subscriptionSetOf.addListener(object : EventListener() {
            override fun message(pubnub: PubNub, pnMessageResult: PNMessageResult) {
                println("-= message for subscriptionSetOf=-")
                println(pnMessageResult)
                success.set(true)
            }
        })

        subscriptionSetOf.subscribe()



        guestClient.publish(
            channel = randomChannelName01,
            message = expectedMessage + "01"
        ).sync()!!
        guestClient.publish(
            channel = randomChannelName02,
            message = expectedMessage + "02"
        ).sync()!!

        success.listen(20)
    }
```